src/utils/validators.js

```diff
@@ -218,6 +240,57 @@ const validateHighlights = (highlightError, field, value) => {
   return newHighlightError
 }
 
+const validateAnnouncementItems = (announcementError, field, value) => {
```

Comment:
could we separate out the validation from the set? this makes it easier to reuse in the future, which was 1 blocker i faced when trying to use the existing validators

Reply:
validation from the set? May I get your help to expand more on this/direct me to the part of the codebase that uses the pattern you are trying to explain?

Reply:
abit unclear wording on my end, but `validateAnnouncementItems` sets the corresponding index. in this case, it does **not** simply validate (ie, return bool + msg) but also does a side effect (setting the corresponding index). this isn't particularly important to fix _now_ because it follows the existing style of code but when we want to do simple validation that is unaware of the _structure_ of the error, will be an issue

Reply:
hmm ok, am inclined to keep it as is for consistency with the rest of the local code, will modify if future pain points warrant it?